pkg/bootkube/bootstrap.go

```diff
@@ -0,0 +1,84 @@
+package bootkube
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/kubernetes-incubator/bootkube/pkg/asset"
+)
+
+// CreateBootstrapControlPlane seeds static manifests to the kubelet to launch the bootstrap control
+// plane.
+func CreateBootstrapControlPlane(assetDir string, podManifestPath string) error {
+	UserOutput("Running temporary bootstrap control plane...\n")
+
+	// Make secrets temporarily available to bootstrap cluster.
+	if err := os.RemoveAll(asset.BootstrapSecretsDir); err != nil {
+		return err
+	}
+	if err := os.Mkdir(asset.BootstrapSecretsDir, os.FileMode(0755)); err != nil {
+		return err
+	}
+	secretsDir := filepath.Join(assetDir, asset.AssetPathSecrets)
+	secrets, err := ioutil.ReadDir(secretsDir)
+	if err != nil {
+		return err
+	}
+	for _, secret := range secrets {
+		if err := copyFile(filepath.Join(secretsDir, secret.Name()), filepath.Join(asset.BootstrapSecretsDir, secret.Name()), true); err != nil {
+			return err
+		}
+	}
+
+	// Copy the static manifests to the kubelet's pod manifest path.
+	manifestsDir := filepath.Join(assetDir, asset.AssetPathBootstrapManifests)
+	manifests, err := ioutil.ReadDir(manifestsDir)
+	if err != nil {
+		return err
+	}
+	for _, manifest := range manifests {
+		if err := copyFile(filepath.Join(manifestsDir, manifest.Name()), filepath.Join(podManifestPath, manifest.Name()), false); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// CleanupBootstrapControlPlane brings down the bootstrap control plane and cleans up the temporary
+// secrets.
+func CleanupBootstrapControlPlane(assetDir string, podManifestPath string) error {
+	UserOutput("Cleaning up temporary bootstrap control plane...\n")
+
+	manifests, err := ioutil.ReadDir(filepath.Join(assetDir, asset.AssetPathBootstrapManifests))
+	if err != nil {
+		return err
+	}
+	for _, manifest := range manifests {
+		if err := os.Remove(filepath.Join(podManifestPath, manifest.Name())); err != nil {
+			return err
+		}
+	}
+	if err := os.RemoveAll(asset.BootstrapSecretsDir); err != nil {
```

Comment:
Agreed. Switched the order to remove the secrets first. Overall the chaining of all this is a little brittle. I'll give some thought to a smarter way for handling the shared secrets.